src/modules/get_professors_by_class/app/get_professors_by_class_controller.ts

```diff
@@ -0,0 +1,31 @@
+
+import { MissingParameters } from '../../../shared/helpers/errors/controller_errors';
+import { IRequest, IResponse } from '../../../shared/helpers/external_interfaces/external_interface';
+import { GetProfessorsByClassUsecase } from './get_professors_by_class_usecase';
+import { OK, BadRequest, NotFound ,  InternalServerError} from '../../../shared/helpers/external_interfaces/http_codes';
+import { NoItemsFound } from '../../../shared/helpers/errors/usecase_errors';
+import { GetProfessorsByClassViewModel } from './get_professors_by_class_viewmodel';
+
+export class GetProfessorsByClassController {
+  constructor(private usecase: GetProfessorsByClassUsecase) {}
+
+  async execute(request: IRequest): Promise<IResponse> {
+    try {
+      const classId = request.data.classId as string;
+      if (!classId) {
+        throw new MissingParameters('classId');
+      }
```

Comment:
call Class.validateId() method (return error `WrongTypeParameters` if the type is wrong)